### DISCUSSION TITLE
Single value for summation objective function

### DISCUSSION THREAD

**Discussion by Dlee0919** [coin-or/Ipopt#659] [Q&A] [answered]
Hello, now I need to solve the objective function of this form :
f = w_1 * g_1(x) + w2 * g_2(x)
and I want to know the value of single item ( e.g. w_1*g_1(x) ) of the function in each iteration, but I can't find the code or options to do this.

**Comment by svigerske** [comment, marked as answer]
I don't see how Ipopt could help on this. It gets the value of the objective function (f) from the user and doesn't know about each term in it.

**Reply by Dlee0919** [reply]
Thank you for your reply! I have another question: Whether Ipopt can return the value of the solution(x) in each iteration?

**Reply by svigerske** [reply]
Ipopt calls the "intermediate callback" at every iteration. You can access the current solution at that point.
https://coin-or.github.io/Ipopt/INTERFACES.html#INTERFACE_CPP_ADDITIONAL
